src/wallet/coinselection.cpp

```diff
@@ -448,6 +448,13 @@ util::Result<SelectionResult> CoinGrinder(std::vector<OutputGroup>& utxo_pool, c
                 best_selection_weight = curr_weight;
                 best_selection_amount = curr_amount;
             }
+        } else if (!best_selection.empty() && curr_weight + min_tail_weight[curr_tail] * ((total_target - curr_amount) / utxo_pool[curr_tail].GetSelectionAmount()) > best_selection_weight) {
```

Comment:
I used the suggestion to upcast and to round up, but I’m leaving the `!best_selection.empty()` in, as we would otherwise not produce a `max_tx_weight_exceeded` error.